bundle/jsky.app.ot/src/main/scala/jsky/app/ot/viewer/open/OpenDialog.scala

```diff
@@ -202,6 +216,55 @@ class OpenDialog private(db: IDBDatabaseService, auth: KeyChain, vcs: VcsRegistr
       dialog.close()
     }
 
+    lazy val RevertAction = Action("Revert") {
+      val ps    = Contents.ProgTable.selectedProgs.filter(Contents.ProgTable.hasPendingChanges)
+      val total = ps.size
+
+      val warning: String =
+        s"""This action will delete all local changes made
+           |since the last sync and restore the program${if (total === 1) "" else "s"}
+           |from the remote database.
+           |
+           |Continue?
+         """.stripMargin
+
+      def doRevert(client: VcsOtClient): Unit = {
+        val ps0 = ps.flatMap { p =>
+          (for {
+            pid  <- Option(p.getProgramID)
+            peer <- client.peer(pid)
+          } yield (pid, peer, p)).toList
+        }
+
+        val result: Throwable \/ List[String \/ Unit] =
+          ps0.traverseU { case (pid, peer, p) =>
+            client.revert(pid, peer, new AtomicBoolean(false)).run
+                  .map(_.bimap(explainFailure(pid, peer, _, "revert"), _ => ()))
+          }.unsafePerformSyncAttempt
+
+        result match {
+          case -\/(_)   =>
+            val msg = "There was an error reverting programs from the remote database."
+            Dialog.showMessage(Contents, msg, "Error", Dialog.Message.Error)
+
+          case \/-(lst) =>
+            val (fs, ss) = lst.separate
+            if (ss.nonEmpty) {
+              Contents.ProgTable.refresh(full = false)
+              updateStorage()
+            }
+            handleFailure(fs)
+        }
+      }
+
+      if (total > 0) {
+        val confirm = Dialog.showConfirmation(dialog.Contents, warning, "Confirm Revert", Dialog.Options.YesNo, Dialog.Message.Warning)
+        if (confirm == Dialog.Result.Ok) {
```

Comment:
It feels weird at this point seeing `==` in Scala code, but I guess trying to define `Eq` on `scala.Enumeration.Value` would be overkill and still not type-safe.

Reply:
Yeah, I just figured it wasn't worth the trouble especially since this code is supposedly slated for retirement in the next few years.